tests/conftest.py

```diff
@@ -0,0 +1,24 @@
+import asyncio
+
+import pytest
+from bluesky.run_engine import RunEngine, TransitionError
+
+
+@pytest.fixture(scope="function")
+def RE(request):
+    loop = asyncio.new_event_loop()
+    loop.set_debug(True)
+    RE = RunEngine({}, call_returns_result=True, loop=loop)
+
+    def clean_event_loop():
+        if RE.state not in ("idle", "panicked"):
+            try:
+                RE.halt()
+            except TransitionError:
+                pass
+        loop.call_soon_threadsafe(loop.stop)
+        RE._th.join()
+        loop.close()
+
+    request.addfinalizer(clean_event_loop)
+    return RE
```

Comment:
A general comment for future reference; I've started peppering this into every repo that needs to run plans with ophyd-async devices, such as dodal, and ophyd-async. And now here! Seems like this bit of code really ought to belong somewhere more common but I'm not sure where. This is just a general comment though. Happy for this to be a 'we will sort it after I22' type deal.

Reply:
Is it possible to have fixtures available from the library (ophyd_async) to libraries (this, dodal) and applications (BlueAPI?) that use it?

Does ophyd_async depend on bluesky for things other than the RunEngine?

Reply:
Ophyd-async mainly depends on bluesky for its protocols (as well as the run engine).

For now I'm thinking the best thing is to have a `testutils` module in bluesky which contains this, i.e.
```
bluesky/bluesky/utils.py::
def RE(request):
    loop = asyncio.new_event_loop()
    loop.set_debug(True)
    RE = RunEngine({}, call_returns_result=True, loop=loop)

    def clean_event_loop():
        if RE.state not in ("idle", "panicked"):
            try:
                RE.halt()
            except TransitionError:
                pass
        loop.call_soon_threadsafe(loop.stop)
        RE._th.join()
        loop.close()

    request.addfinalizer(clean_event_loop)
    return RE
```

And then here,

```
from bluesky.utils import RE

@pytest.fixture(scope="function")
def RunEngine(request):
    return RE(request)
```

but idk if `RE` is a good name for it...

Reply:
Protocols